packages/std/src/imports.rs

```diff
@@ -107,17 +107,21 @@ impl Storage for ExternalStorage {
         let key_ptr = &*key as *const Region as *const c_void;
         let mut value = build_region(value);
         let value_ptr = &mut *value as *mut Region as *mut c_void;
-        unsafe {
-            write_db(key_ptr, value_ptr);
+        let result = unsafe { write_db(key_ptr, value_ptr) };
+        if result < 0 {
+            // TODO: convert to Result, but this requires changing the trait
```

Comment:
Want to open an issue for this?
It is more correct, but it would be good to ensure it has as minimal overhead as possible for the user.
`Result<Option<Vec<u8>>>`?

Reply:
Ticket created: https://github.com/CosmWasm/cosmwasm/issues/216